benchbuild/experiment.py

```diff
@@ -46,25 +46,17 @@ class HelloExperiment(Experiment):
 Projects = tp.List[ProjectT]
 
 
-class ExperimentRegistry(type):
-    """Registry for benchbuild experiments."""
+class NamedEntity:  # pylint: disable=too-few-public-methods
```

Comment:
Not sure if this is the best name. Should we not include something "Experiment" specific into the name?
Is a project an named Entity? Or a container etc...

Reply:
Not included in this PR but this way we could just share the name propery with projects (some refactoring included).
I want to go with a simple 'name' but that one collides too easily with other modules like ABC.